src/components/body/header/Header.tsx

```diff
@@ -1,69 +1,92 @@
 import "./Header.scss";
 import {FormattedMessage, useIntl} from "react-intl";
-import Avatar from "../../icons/Avatar.svg";
+import {Link, useNavigate} from "react-router-dom";
+import {showDefaultPage, showStudentPage} from "../../../store/header/headerSlice";
+import {useAppDispatch, useAppSelector} from "../../../store/hooks";
+import Avatar from "./Avatar";
 import Bell from "../../icons/Bell.svg";
 import Button from "../../Button";
-import {Link} from "react-router-dom";
 import Logo from "../../icons/Logo.svg";
 import Magnifier from "../../icons/blackMagnifier.svg";
-import {useAppSelector} from "../../../store/hooks";
+import {nameDecodeUser} from "../../../store/header/decodeJwtSlice";
+import {useEffect} from "react";
 
 const Header = () => {
   const intl = useIntl();
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
   const isDefaultPage = useAppSelector(state => state.value.isDefaultHeader);
   const page = useAppSelector(state => state.value.page);
-
+  const decodeUserName = useAppSelector(state => state.userDecodeName.name);
   const userButtonText = page === "sectionPage" ? intl.formatMessage({id: "app.header.myStudio"})
     : intl.formatMessage({id: "app.header.goStudy"});
+  const loading = useAppSelector(state => state.login.loading);
+
+  useEffect(() => {
+    dispatch(nameDecodeUser());
+    if (decodeUserName) {
+      navigate("");
```

Comment:
navigate(""); is the same despite if .. else... 

replace it from if... else... construction